### ISSUE TITLE
disable-gpu-compositing always present when using headless mode

### ISSUE THREAD

**Issue by danhoss** [puppeteer/puppeteer#6083]
**TL;DR:**

When using `--headless` flag puppeteer always also sets (implicitly) `--disable-gpu-compositing` `--allow-pre-commit-input` flags

Browser: HeadlessChrome/83.0.4103.106
OS: Linux 4.13.0-46-generic

Don't know if this is actual bug, but didn't find anywhere on the Internet info about why that happens or how to remove this flag on headless.

--------------------------------------------------------------------------------------------------------------

**Use case:**
Im using puppeteer to generate PNG and PDF files from html source containing svg element.

Problem occurs when svg contains huge images (e.g. png image of size 12.000x15.000px)

When using window mode whole process of generating pdf/png takes about 45 seconds (not great, not terrible), but by adding just one option `--headless` time increases to over 30 minutes(!).

After hours of investigation, tweaking and adjusting settings it came out that with `--headless` flag puppeteer also sets (implicitly) `--disable-gpu-compositing` `--allow-pre-commit-input` flags, and due to this compositing is "_Software only. Hardware acceleration disabled_". I found no way to turn this off (means to enable gpu compositioning).

**Comment by booxood** [comment]
The same thing happened to me.

puppeteer: 1.20.0
Chromium: 78.0.3882.0

**Comment by yuvraj-pyxis** [comment]
I have been experiencing the same issue. Can we get any update to how can we explicitly allow compositing to be hardware accelerated?

**Comment by net** [comment]
Chromium/Chrome requires `--use-gl=egl` to [enable GPU acceleration in headless mode](https://github.com/chromium/chromium/commit/19671359ae25aa1e30bde90f8ff92453eeaac2ba).

```js
puppeteer.launch({ headless: true, args: ['--use-gl=egl'] });
```

**Comment by vJoeyz** [comment]
still having this issue on Windows, `--use-gl=egl` doesn't seem to work.
